src/components/paster.ts

```diff
@@ -0,0 +1,220 @@
+            }
+        })
+    }
+
+    public pasteFile(editor: vscode.TextEditor, baseFile: string, file: string) {
+        const TABLE_FORMATS = ['.csv']
+        const extension = path.extname(file)
+
+        if (TABLE_FORMATS.indexOf(extension) !== -1) {
+            const contents = fs.readFileSync(path.resolve(baseFile, file), 'utf8')
+            if (extension === '.csv') {
+                // from: https://stackoverflow.com/a/41563966/3026698
+                let p = ''
+                let row = ['']
+                let i = 0
+                let r = 0
+                let s = !0
+                let l
+                const ret = [row]
+                for (l of contents) {
+                    if ('"' === l) {
+                        if (s && l === p) {
+                            row[i] += l
+                        }
+                        s = !s
+                    } else if (',' === l && s) {
+                        l = row[++i] = ''
+                    } else if ('\n' === l && s) {
+                        if ('\r' === p) {
+                            row[i] = row[i].slice(0, -1)
+                        }
+                        row = ret[++r] = [(l = '')]
+                        i = 0
+                    } else {
+                        row[i] += l
+                    }
+                    p = l
+                }
+
+                const rows = ret.map(r => r.join('\t'))
+                const body = rows.join('\n')
+                this.pasteTable(editor, body)
+            }
+        }
+    }
+
+    public pasteTable(editor: vscode.TextEditor, content: string) {
+        this.extension.logger.addLogMessage('Pasting: Table')
+        // trim surrounding whitespace
+        content = content.replace(/^\s*/, '').replace(/\s*$/, '')
+        content = this.reformatText(content, false)
+        const lines = content.split('\n')
+        const cells = lines.map(l => l.split('\t'))
+
+        // determine if all rows have same number of cells
+        const isConsistent = cells.reduce((accumulator, current, _index, array) => {
+            if (current.length === array[0].length) {
+                return accumulator
+            } else {
+                return false
+            }
+        }, true)
+        if (!isConsistent || (cells.length === 1 && cells[0].length === 1)) {
+            throw new Error('Table is not consistent')
+        }
+
+        const columnType: string = vscode.workspace.getConfiguration('latex-workshop.formattedPaste')['tableColumnType']
+        const booktabs: boolean = vscode.workspace.getConfiguration('latex-workshop.formattedPaste')['tableBooktabsStyle']
+        const headerRows: number = vscode.workspace.getConfiguration('latex-workshop.formattedPaste')['tableHeaderRows']
+        const tabularRows = cells.map(row => '\t' + row.join(' & '))
+
+        if (headerRows && tabularRows.length > headerRows) {
+            const eol = editor.document.eol.toString()
```

Comment:
This is wrong. `eol` returns `number`, `1` or `2`. Please read the document carefully.

Reply:
Ok. Sorry about that. Should be fixed in 1d8a6e36efa94817fcabc0c3908271f2bc66678d